### PR TITLE
Improved structure for objectstore unit tests.

### PR DESCRIPTION
**Pull request by aclamk**
Modified structure of matrix-based unit tests.
Now each test case can be invoked separately.
New matrix tests are hard bound to "BlueStore".
    
Now:
    ...
    BlueStore/SyntheticShardingLimited.
      Test/0  # GetParam() = { 65536, 10000, 65536, 262144, 4096, 1000, 65536, 60, 300, 150, true, true }
      Test/1  # GetParam() = { 4096, 10000, 65536, 262144, 4096, 1000, 65536, 60, 300, 150, true, true }
      Test/2  # GetParam() = { 65536, 10000, 65536, 262144, 4096, 200, 65536, 60, 300, 150, true, true }
      Test/3  # GetParam() = { 4096, 10000, 65536, 262144, 4096, 200, 65536, 60, 300, 150, true, true }
      Test/4  # GetParam() = { 65536, 10000, 65536, 262144, 4096, 50, 65536, 60, 300, 150, true, true }
      Test/5  # GetParam() = { 4096, 10000, 65536, 262144, 4096, 50, 65536, 60, 300, 150, true, true }
    BlueStore/SyntheticMatrixCsumAlgorithm.
      Test/0  # GetParam() = { 65536, 65536, 1048576, 16, crc32c, false }
      Test/1  # GetParam() = { 65536, 65536, 1048576, 16, crc32c_16, false }
      Test/2  # GetParam() = { 65536, 65536, 1048576, 16, crc32c_8, false }
      Test/3  # GetParam() = { 65536, 65536, 1048576, 16, xxhash32, false }
      Test/4  # GetParam() = { 65536, 65536, 1048576, 16, xxhash64, false }
      Test/5  # GetParam() = { 65536, 65536, 1048576, 16, none, false }
    BlueStore/SyntheticMatrixCsumVsCompression.
      Test/0  # GetParam() = { 4096, 131072, 262144, 512, force, snappy, crc32c, true, true, false }
      Test/1  # GetParam() = { 16384, 131072, 262144, 512, force, snappy, crc32c, true, true, false }
      Test/2  # GetParam() = { 4096, 131072, 262144, 512, force, zlib, crc32c, true, true, false }
      Test/3  # GetParam() = { 16384, 131072, 262144, 512, force, zlib, crc32c, true, true, false }
    ...
Before:
    ...
      SyntheticLimited/0  # GetParam() = "memstore"
      SyntheticLimited/1  # GetParam() = "bluestore"
      SyntheticLimited/2  # GetParam() = "kstore"
      SyntheticMatrixCsumAlgorithm/0  # GetParam() = "memstore"
      SyntheticMatrixCsumAlgorithm/1  # GetParam() = "bluestore"
      SyntheticMatrixCsumAlgorithm/2  # GetParam() = "kstore"
      SyntheticMatrixCsumVsCompression/0  # GetParam() = "memstore"
      SyntheticMatrixCsumVsCompression/1  # GetParam() = "bluestore"
      SyntheticMatrixCsumVsCompression/2  # GetParam() = "kstore"
    ...

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by aclamk** [comment]
jenkins test make check

**Comment by aclamk** [comment]
jenkins test make check

File: src/test/objectstore/store_test.cc [hunk 1]
```diff
@@ -148,22 +148,28 @@ bool sorted(const vector<ghobject_t> &in) {
...
-    int num_ops,
+    int initial_object_count, int num_ops,
     uint64_t max_obj, uint64_t max_wr, uint64_t align);
   // a variant of test that keeps amount of active objects stable
   void doSyntheticLimitedTest(
-    int initial_object_count,
-    int num_ops,
+    int initial_object_count, int num_ops,
     uint64_t max_obj, uint64_t max_wr, uint64_t align);
+
```
Comment (ronen-fr) [inline src/test/objectstore/store_test.cc:162]: this specific blank line does not add clarity, I think.

File: src/test/objectstore/store_test.cc [hunk 2]
```diff
@@ -197,35 +226,23 @@ class StoreTestSpecificAUSize : public StoreTestDeferredSetup {
     doSyntheticLimitedTest(start_object_count, num_ops, max_size, max_write, alignment);
   }
 
+  void SetUp() override {
+    const MatrixRow row = GetParam();
+    for (auto& i: row) {
```
Comment (ronen-fr) [inline src/test/objectstore/store_test.cc:231]: const auto& would be enough

File: qa/suites/rados/objectstore/backends/objectstore-bluestore-a.yaml [hunk 3]
```diff
@@ -8,5 +8,5 @@ tasks:
 - install:
 - exec:
     client.0:
-      - mkdir $TESTDIR/archive/ostest && cd $TESTDIR/archive/ostest && ulimit -Sn 16384 && CEPH_ARGS="--no-log-to-stderr --log-file $TESTDIR/archive/ceph_test_objectstore.log --debug-bluestore 20" ceph_test_objectstore --gtest_filter=*/1:-*SyntheticMatrixC* --gtest_catch_exceptions=0
+      - mkdir $TESTDIR/archive/ostest && cd $TESTDIR/archive/ostest && ulimit -Sn 16384 && CEPH_ARGS="--no-log-to-stderr --log-file $TESTDIR/archive/ceph_test_objectstore.log --debug-bluestore 20" ceph_test_objectstore --gtest_filter=*/1:-*Matrix*Compression* --gtest_catch_exceptions=0
```
Comment (ifed01) [inline qa/suites/rados/objectstore/backends/objectstore-bluestore-a.yaml:11]: IMO new filter will filter out most of matrix tests (except ones ending with /1) which is different from the original behavior.

So the proper filter should be:
--gtest_filter=*/1:*Matrix*:-*Matrix*Compression*

**Review by ifed01 — CHANGES_REQUESTED** [review]

File: qa/suites/rados/objectstore/backends/objectstore-bluestore-a.yaml (hunk 3 above)
Comment (aclamk) [reply qa/suites/rados/objectstore/backends/objectstore-bluestore-a.yaml:11]: RIGHT!

Comment (aclamk) [reply qa/suites/rados/objectstore/backends/objectstore-bluestore-a.yaml:11]: Right, fixed.

**Review by ifed01 — APPROVED** [review]

**Comment by aclamk** [comment]
jenkins test api

**Comment by aclamk** [comment]
jenkins test api

**Comment by aclamk** [comment]
This PR passed rados suite.
https://pulpito.ceph.com/akupczyk-2024-07-12_08:52:40-rados-aclamk-testing-ganymede-2024-07-10-1144-distro-default-smithi/
https://pulpito.ceph.com/akupczyk-2024-07-11_11:24:14-rados-aclamk-testing-ganymede-2024-07-10-1144-distro-default-smithi/

But there was no bluestore-a/bluestore-b tests.

**Comment by aclamk** [comment]
https://pulpito.ceph.com/akupczyk-2024-07-19_20:17:10-rerun-aclamk-testing-ganymede-2024-07-10-1144-distro-default-smithi/
bluestore a/b tested.

**Pull request merged**